kb(wizard): confirm moving between steps (#856)
telerik/blazor-docs: knowledge-base/wizard-confirm-step-change.md

---

> title: How do I pop up a confirmation window or dialog between Wizard steps?

Reviewer: ```suggestion
title: Confirm between Wizard steps
```

> description: How to show a dialog confirming the change of the wizard steps

Reviewer: ```suggestion
description: How to show a popup dialog or window to confirm the change of the current Wizard step.
```

type: how-to

> page_title: Confirm between Wizard steps

Reviewer: The Page Title doesn't have to be that short.

```suggestion
page_title: How to pop up a confirmation dialog between Wizard steps
```

slug: wizard-kb-confirm-step-change
position: 
tags: wizard
ticketid: 1559188
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Wizard for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

> When a user wants to move to the next step, I need to confirm the move (e.g. show them a warning for some situations), and if they choose to "continue" take them to the next step as normal, not back to the current step.

Reviewer: ```suggestion
When a user wants to move to the next step, I need to confirm the move and show them a popup warning for some situations. If they choose to "continue", take them to the next step as normal, otherwise back to the current step.
```

## Solution

> Use the [OnChange]({%slug wizard-events%}#onchange) event of the originating step to show a [confirm dialog]({%slug dialog-predefined%}#confirm) and cancel the event if the user does not confirm.

Reviewer: ```suggestion
Use the [`OnChange` event]({%slug wizard-events%}#onchange) of the originating step to show a [confirm Dialog]({%slug dialog-predefined%}#confirm). Cancel the event if the user does not confirm. It is also possible to use a Window component, but a confirmation Dialog is likely more suitable.
```

> This example shows confirmation when moving away from step 1, you can extend it for more/other steps. You can also add a lambda expression in the handler on each step to provide some more metadata to the method - for example, to form a more meaningful message for the step, to discern steps more easily, or any other business need.

Reviewer: ```suggestion
This example shows confirmation when moving away from step 1. You can extend it for other steps. You can also add a lambda expression in the handler on each step to provide more data to the method - for example, to form a more meaningful message for the step or to distinguish steps more easily.
```
